kv/dist_sender_server_test.go

```diff
@@ -83,9 +82,14 @@ func setupMultipleRanges(t *testing.T, splitAt ...string) (*server.TestServer, *
 	s := server.StartTestServer(t)
 	db := createTestClient(t, s.Stopper(), s.ServingAddr())
 
+	// Wait for initial splits to finish.
+	s.WaitForInitialSplits(t, time.Second)
+
 	// Split the keyspace at the given keys.
 	for _, key := range splitAt {
 		if err := db.AdminSplit(key); err != nil {
+			// Don't leak server goroutines.
+			defer s.Stop()
```

Comment:
This is a weird place to defer a call to `s.Stop()` since it's far from the server creation. I'd call it without `defer` here.